cdc/processor/sinkmanager/manager.go

```diff
@@ -185,7 +186,12 @@ func (m *SinkManager) Run(ctx context.Context, warnings ...chan<- error) (err er
 	}()
 
 	splitTxn := util.GetOrZero(m.changefeedInfo.Config.Sink.TxnAtomicity).ShouldSplitTxn()
-	enableOldValue := m.changefeedInfo.Config.EnableOldValue
+
+	var shouldSplitUpdate bool
```

Comment:
Please move this check into the `config` struct. Then we don't need to leak that information here. Maybe we only need a method called `EnableOldValue()`. Then we can play the trick inside the config.

What happened if we use the MySQL sink and add a protocol here?